File: Formula-Bollo-App/src/main/java/formula/bollo/app/entity/Team.java
package formula.bollo.app.entity;

import java.sql.Blob;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.GenerationType;

@Entity
@Table(name = "team")
@Getter
@Setter
@ToString
@EqualsAndHashCode
@AllArgsConstructor
@NoArgsConstructor
public class Team {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false, length = 50)
    @NotNull
    private String name;

    @Column(name = "car_image", nullable = false)
    @NotNull
    private String carImage;
    
    @JsonIgnore
    @Column(name = "logo_image", nullable = false, columnDefinition = "BLOB")
    @NotNull
    private Blob logoImage;

    @ManyToOne
    @JoinColumn(name = "season_id")
    private Season season;
}

File: Formula-Bollo/src/shared/models/team.ts
import { Season } from "./season";

export class Team {
  id: number;
  name: string;
  carImage: string;
  logoImage: string;
  season: Season;

  constructor(
    id: number,
    name: string,
    carImage: string,
    logoImage: string,
    season: Season,
  ) {
    this.id = id;
    this.name = name;
    this.carImage = carImage;
    this.logoImage = logoImage;
    this.season = season;
  }
}
